### PR TITLE
Enable gzip for static files for webgui

### PR DESCRIPTION
**Pull request by ralder**

### CODE REVIEW

File: cmd/genassets/main.go [hunk 1]
```diff
@@ -15,33 +15,32 @@ import (
...
 
 import (
-	"bytes"
-	"compress/gzip"
 	"encoding/base64"
-	"io/ioutil"
+)
+
+const (
+	ASSETS_BUILD_DATE = "` + time.Now().UTC().Format(http.TimeFormat) + `"
```
Comment (calmh) [inline cmd/genassets/main.go]: `assetsBuildDate`

Comment (calmh) [inline cmd/genassets/main.go]: But plopping in the string in the template like this is not beautiful. Better make it a variable referenced in the template?

**Comment by calmh** [comment]
Tests need tweaking. 

Do we need to send a not acceptable response in the unlikely case of a browser not announcing `Accept-Encoding: gzip`?

File: cmd/genassets/main.go (hunk 1 above)
Comment (ralder) [inline cmd/genassets/main.go]: go need uppercase first letter in name to export it from package. And in file syncthing/internal/symlinks/symlink_windows.go constants are uppercase, way i can't?

Comment (AudriusButkevicius) [inline cmd/genassets/main.go]: That's win32 api constants, not go constants.

Comment (calmh) [inline cmd/genassets/main.go]: Those are presumably copies of identically named Windows constants, but even that is borderline. Go does not use all caps or underscores. `AssetsBuildTime` is fine if it should be exported. https://github.com/golang/go/wiki/CodeReviewComments#mixed-caps

**Comment by calmh** [comment]
LGTM. Can you squash this to a single commit please?

**Comment by ralder** [comment]
done

**Comment by ralder** [comment]
About browsers which not announcing "Accept-Encoding:gzip" I think all ones which support jquery 2.x will send this header, anyway i added code for handle this case, but seems it will be execute never in 99.9% (and i tested on chrome and firefox send empty header "accept-encoding" , and it worked with gzip)

**Comment by calmh** [comment]
Yeah, as far as I know all current browsers do support it, so I was really thinking just send back an error if they don't, but this is fine too.

**Comment by calmh** [comment]
Actually, `curl` is one of those non-supporting things apparently, so this is good.

**Comment by calmh** [comment]
@st-jenkins run integration tests, please

**Comment by calmh** [comment]
The `TestGetIndex` failure in the integration tests looks genuine (there's also a spurious failure or two in there not related to this PR).

**Comment by calmh** [comment]
I though so too, but my gofmt doesn't seem to mind. :)

```bash
jb@syno:~/src/github.com/syncthing/syncthing ((detached) $ gofmt -w -l cmd/syncthing/*.go
jb@syno:~/src/github.com/syncthing/syncthing ((detached) $ grep ungz cmd/syncthing/*.go
cmd/syncthing/gui.go:			//ungzip if browser not send gzip accepted header
```

**Comment by AudriusButkevicius** [comment]
Lgtm

**Comment by calmh** [comment]
That `TestGetIndex` failure needs fixing though, I'm not sure why it would see a -1 content-length, but it's a bad sign.

**Comment by calmh** [comment]
@st-jenkins run integration tests

**Comment by calmh** [comment]
@st-jenkins run integration tests

**Comment by ralder** [comment]
I'll try fix it

**Comment by ralder** [comment]
It's go lang http.Client feature. He send Accept-Encoding:gzip by default and automatic ungzip it if reading response.Body but before reading he not knows real content length and set it to -1

**Comment by calmh** [comment]
Okay. Maybe do a ReadAll or something on that to verify we actually get a reasonably sized html file back then...

**Comment by ralder** [comment]
@st-jenkins run integration tests

**Comment by calmh** [comment]
:+1:

**Pull request merged**